app/src/main/java/org/wikipedia/appshortcuts/AppShortcuts.kt

```diff
@@ -22,11 +22,12 @@ class AppShortcuts {
         private const val APP_SHORTCUT_ID_CONTINUE_READING = "shortcutContinueReading"
         private const val APP_SHORTCUT_ID_RANDOM = "shortcutRandom"
         private const val APP_SHORTCUT_ID_SEARCH = "shortcutSearch"
+        private const val APP_SHORTCUT_ID_PLACES = "shortcutPlaces"
 
         fun setShortcuts(app: Context) {
             CoroutineScope(Dispatchers.Default).launch(CoroutineExceptionHandler { _, msg -> run { L.e(msg) } }) {
-                val list = listOf(searchShortcut(app), continueReadingShortcut(app), randomShortcut(app))
-                if (ShortcutManagerCompat.getDynamicShortcuts(app).size < list.size) {
+                val list = listOf(searchShortcut(app), continueReadingShortcut(app), randomShortcut(app), placesShortcut(app))
+                if (ShortcutManagerCompat.getDynamicShortcuts(app).containsAll(list).not()) {
```

Comment:
This check is no longer correct. The object instances in the local `list` are always different from what's returned by `getDynamicShortcuts()`, so this will always be true, and will always unnecessarily recreate the dynamic shortcuts.
If we really want to do the check correctly, we should check if the shortcut `id`s match.

Reply:
Done. I have noticed that the app shortcut function only allows `4` shortcuts, so the `LeakCanary` app shortcut will be removed after we reset the shortcuts.